src/react/hooks/paragon/useParagonThemeCore.js

```diff
@@ -0,0 +1,144 @@
+import { useEffect, useState } from 'react';
+
+import { logError, logInfo } from '../../../logging';
+import { fallbackThemeUrl, removeExistingLinks } from './utils';
+
+/**
+ * Adds/updates a `<link>` element in the HTML document to load the core application theme CSS.
+ *
+ * @memberof module:React
+ *
+ * @param {object} args
+ * @param {object} args.themeCore Object representing the core Paragon theme CSS.
+ * @param {string} args.onLoad A callback function called when the core theme CSS is loaded.
+ */
+const useParagonThemeCore = ({
+  themeCore,
+  onLoad,
+}) => {
+  const [isParagonThemeCoreLoaded, setIsParagonThemeCoreLoaded] = useState(false);
+  const [isBrandThemeCoreLoaded, setIsBrandThemeCoreLoaded] = useState(false);
+
+  useEffect(() => {
+    // Call `onLoad` once both the paragon and brand theme core are loaded.
+    if (isParagonThemeCoreLoaded && isBrandThemeCoreLoaded) {
+      onLoad();
+    }
+  }, [isParagonThemeCoreLoaded, isBrandThemeCoreLoaded, onLoad]);
+
+  useEffect(() => {
+    // If there is no config for the core theme url, do nothing.
+    if (!themeCore?.urls) {
+      setIsParagonThemeCoreLoaded(true);
+      setIsBrandThemeCoreLoaded(true);
+      return;
+    }
+    const getParagonThemeCoreLink = () => document.head.querySelector('link[data-paragon-theme-core="true"]');
+    const existingCoreThemeLink = document.head.querySelector(`link[href='${themeCore.urls.default}']`);
+    const brandCoreLink = document.head.querySelector(`link[href='${themeCore.urls.brandOverride}']`);
+    if (!existingCoreThemeLink) {
+      const getExistingCoreThemeLinks = (isBrandOverride) => {
+        const coreThemeLinkSelector = `link[data-${isBrandOverride ? 'brand' : 'paragon'}-theme-core="true"]`;
+        return document.head.querySelectorAll(coreThemeLinkSelector);
+      };
+      const createCoreThemeLink = (
+        url,
+        {
+          isFallbackThemeUrl = false,
+          isBrandOverride = false,
+        } = {},
+      ) => {
+        let coreThemeLink = document.createElement('link');
+        coreThemeLink.href = url;
+        coreThemeLink.rel = 'stylesheet';
+        if (isBrandOverride) {
+          coreThemeLink.dataset.brandThemeCore = true;
+        } else {
+          coreThemeLink.dataset.paragonThemeCore = true;
+        }
+        coreThemeLink.onload = () => {
+          if (isBrandOverride) {
+            setIsBrandThemeCoreLoaded(true);
+          } else {
+            setIsParagonThemeCoreLoaded(true);
+          }
+        };
+        coreThemeLink.onerror = () => {
+          logError(`Failed to load core theme CSS from ${url}`);
+          if (isFallbackThemeUrl) {
+            logError(`Could not load core theme CSS from ${url} or fallback URL. Aborting.`);
+            if (isBrandOverride) {
+              setIsBrandThemeCoreLoaded(true);
+            } else {
+              setIsParagonThemeCoreLoaded(true);
```

Comment:
[curious] Do we want to ensure coverage on the critical code paths within `useParagonThemeCore`, e.g. around loading? There's a few CodeCov annotations about missing coverage.

Reply:
Yes, working on the overall coverage :)